Dockerfile

```diff
@@ -62,4 +84,57 @@ RUN set -ex \
       && cd ../ \
       && rm -rf pam_script-$PAM_SCRIPT_VERSION \
       && rm -rf pam_script.zip \
-      && apk del .build-deps;
\ No newline at end of file
+      && apk del .build-deps;
+
+
+ADD rootfs /
+
+ADD https://raw.githubusercontent.com/cloudposse/build-harness/master/templates/Makefile.build-harness Makefile
+
+RUN if [ ! -z $GITHUB_TOKEN ]; then \
+      set -ex \
+      && echo "http://dl-cdn.alpinelinux.org/alpine/edge/testing" >> /etc/apk/repositories \
+      && apk update \
+      && apk add --no-cache --virtual .build-deps \
+          curl \
+          jq \
+          pamtester \
+          make \
+          git \
+      && make  \
+      && REPO=cloudposse/github-pam \
+          VERSION=$GITHUB_PAM_VERSION \
+          FILE=github-pam_linux_386 \
+          OUTPUT=github-pam-plugin \
+          make github:download-release  \
+      && chmod +x github-pam-plugin \
+      && mv github-pam-plugin /bin/ \
+      && apk add ca-certificates \
+      && apk del .build-deps; \
+    else \
+      echo '`GITHUB_TOKEN` required for fetching github-pam-plugin'; \
+      exit 1; \
+    fi
+
+RUN if [ ! -z $GITHUB_TOKEN ]; then \
+      set -ex \
+      && apk update \
+      && apk add --no-cache --virtual .build-deps \
+          curl \
+          jq \
+          git \
+          make \
+      && make \
+      && REPO=cloudposse/openvpn-api \
+          VERSION=$OPENVPN_API_VERSION \
+          FILE=openvpn-api_linux_386 \
+          OUTPUT=openvpn-api \
+          make github:download-release \
+      && chmod +x openvpn-api \
+      && mv openvpn-api /bin/ \
+      && apk del .build-deps; \
+    else \
+      echo '`GITHUB_TOKEN` required for fetching openvpn-api'; \
```

Comment:
Maybe we can modify `github:download-release` to `github:download-private-release` and `github:download-public-release`